### PR TITLE
Feature/marketing/landing pages (ED-378)

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Quality assurance

- [X] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: tests/phpunit/elementor/modules/landing-pages/documents/landing-page.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (kobizz) [inline tests/phpunit/elementor/modules/landing-pages/documents/landing-page.php:1]: Remove this test file

File: tests/phpunit/elementor/modules/landing-pages/test-module.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace Elementor\Testing\Modules\LandingPages;
+
+use Elementor\Testing\Elementor_Test_Base;
+use Elementor\Modules\LandingPages\Module;
+
+class Elementor_Test_Landing_Pages_Module extends Elementor_Test_Base {
+	/** @var Module */
+	private static $module;
+
+	public function test_should_confirm_module_activation() {
```
Comment (kobizz) [inline tests/phpunit/elementor/modules/landing-pages/test-module.php:11]: Change this method name and don't initialize a new module

File: tests/phpunit/elementor/modules/landing-pages/test-module.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		// Initialize an instance of the module being tested.
+		self::$module = new Module();
+
+		$this->assertDocumentTypeRegistered( Module::DOCUMENT_TYPE );
+	}
+
+	public function test_should_return_landing_pages() {
```
Comment (kobizz) [inline tests/phpunit/elementor/modules/landing-pages/test-module.php:18]:
```suggestion
	public function test_get_name() {
```

File: tests/phpunit/elementor/modules/landing-pages/test-module.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		$this->assertEquals( 'landing-pages', self::$module->get_name() );
+	}
+
+	// TODO: move the following method to the documents trait.
+	/**
+	 * Asserts that a type is registered in documents
+	 *
+	 * @param string $type_name
+	 */
+	private function assertDocumentTypeRegistered( $type_name ) {
```
Comment (kobizz) [inline tests/phpunit/elementor/modules/landing-pages/test-module.php:28]: Move this method to a more general location

File: tests/phpunit/elementor/modules/library/test-module.php [hunk 5]
```diff
@@ -19,6 +19,7 @@ public function test_should_return_library() {
 		$this->assertEquals( 'library', self::$module->get_name() );
 	}
 
+	// TODO: move the following method to the documents trait.
```
Comment (kobizz) [inline tests/phpunit/elementor/modules/library/test-module.php:22]: Remove this comment

**Review by kobizz — CHANGES_REQUESTED** [review]

File: modules/landing-pages/assets/js/admin/landing-pages.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+	getDefaultSettings() {
+		const slugs = {
+				landingPagesTablePage: 'edit.php?post_type=page&elementor_library_type=landing-page',
+				landingPagesAddNewPage: 'admin.php?page=landing-page',
+			},
+			adminMenuSelectors = {
+				landingPagesTablePage: '.toplevel_page_edit\\?post_type\\=page\\&elementor_library_type\\=landing-page',
+				landingPagesAddNewPage: '.toplevel_page_landing-page',
+			},
+			landingPagesMenuItemSlug = elementorAdminConfig?.landingPagesHasPages ? slugs.landingPagesTablePage : slugs.landingPagesAddNewPage,
```
[the remark this answers is not in the document]
Comment (udidol) [reply modules/landing-pages/assets/js/admin/landing-pages.js]: Done.

File: modules/landing-pages/assets/js/admin/landing-pages.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+			landingPagesMenuItemSelector = elementorAdminConfig?.landingPagesHasPages ? adminMenuSelectors.landingPagesTablePage : adminMenuSelectors.landingPagesAddNewPage;
```
[the remark this answers is not in the document]
Comment (udidol) [reply modules/landing-pages/assets/js/admin/landing-pages.js]: Done.

File: modules/landing-pages/assets/js/admin/landing-pages.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+		return {
+			selectors: {
+				addButton: '.page-title-action:first',
+				badLandingPageAdminMenuLink: 'a[href="edit-tags.php?taxonomy=e-campaigns&page=landing-page"]',
```
[the remark this answers is not in the document]
Comment (udidol) [reply modules/landing-pages/assets/js/admin/landing-pages.js]: Done.

File: modules/landing-pages/assets/js/admin/landing-pages.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	onInit() {
+		super.onInit();
+
+		// If the current page is the Landing Pages Table page (the posts table displaying landing pages).
+		if ( window.location.href.includes( '?post_type=page&elementor_library_type=landing-page' ) ) {
+			// Make sure the active admin top level menu item is 'Campaigns', and not 'Pages'.
+			this.highlightLandingPagesMenu();
+
+			// If there are landing pages, highlight the Landing Pages sub menu item.
+			if ( elementorAdminConfig.landingPagesHasPages ) {
```
[the remark this answers is not in the document]
Comment (udidol) [reply modules/landing-pages/assets/js/admin/landing-pages.js]: Done.

File: modules/landing-pages/documents/landing-page.php [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+		$properties['show_in_library'] = true;
+
+		return $properties;
+	}
+
+	/**
+	 * @access public
+	 */
+	public function get_name() {
+		return 'landing-page';
```
[the remark this answers is not in the document]
Comment (udidol) [reply modules/landing-pages/documents/landing-page.php]: Done.

File: tests/phpunit/elementor/modules/landing-pages/documents/landing-page.php (hunk 1 above)
Comment (udidol) [reply tests/phpunit/elementor/modules/landing-pages/documents/landing-page.php]: Done.

File: tests/phpunit/elementor/modules/landing-pages/test-module.php (hunk 2 above)
Comment (udidol) [reply tests/phpunit/elementor/modules/landing-pages/test-module.php]: Done.

File: tests/phpunit/elementor/modules/landing-pages/test-module.php (hunk 3 above)
Comment (udidol) [reply tests/phpunit/elementor/modules/landing-pages/test-module.php]: Done.

File: tests/phpunit/elementor/modules/landing-pages/test-module.php (hunk 4 above)
Comment (udidol) [reply tests/phpunit/elementor/modules/landing-pages/test-module.php]: Done. Created a trait called Elementor_Library and put it in `tests/phpunit/traits`

File: tests/phpunit/elementor/modules/library/test-module.php (hunk 5 above)
Comment (udidol) [reply tests/phpunit/elementor/modules/library/test-module.php]: Done.

**Pull request merged**